planet/src/main/webapp/places.js

```diff
@@ -145,29 +158,39 @@ function setSearchByButton() {
 
 /** Create markers and add details to each place */
 function addPlaceDetails() {
-    // For each place, get the icon, name and location.
-    places.forEach(function(place) {
-        if (!place.geometry) {
-          console.log("Returned place contains no geometry");
-          return;
-        }
-        createMarkers(place); 
-        if (place.geometry.viewport) {
-            bounds.union(place.geometry.viewport);
-        } else {
-            bounds.extend(place.geometry.location);
-        }
+    const half = Math.ceil(places.length /2); 
+    const firstHalf = places.splice(0,half);
+    const secondHalf = places.splice(-half);
 
-        // Make request with fields
-        var place = {
-            placeId: place.place_id,
-            fields: ['place_id','name','rating','formatted_phone_number','formatted_address',
-            'opening_hours','photos','url']
-        };
-        // Call Places Details Request
-        let service = new google.maps.places.PlacesService(map);
-        service.getDetails(place, callback); 
+    // For each place, get the icon, name and location.
+    firstHalf.forEach(function(place) {
+        sendPlaceRequest(place);
     });
+
+    // Call remaining 10 place requests after 5 seconds to wait after request query empties
```

Comment:
The documentation implies that requests are re-issued at a rate of 1 queries per second. Have you confirmed that it regenerates at 2 per second?

Reply:
yes! I've tested the settimeout function and the fastest time possible to have all the results go through is 5 seconds